### PR TITLE
Third-party notice for NuGet repo and VSIX

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Home/issues/7409
Regression: No
* Last working version:   N/A
* How are we preventing it in future:   N/A

## Fix

Added Third-Party Notice based on manual inspection of non-Microsoft signed assemblies.

## Testing/Validation

Tests Added: No
Reason for not adding tests:  needs more work for adding automated tests
Validation:  Manual validation

### CODE REVIEW

**Comment by karann-msft** [comment]
@dominoFire - I am not sure if I am the right person to review this.
Let me know if you need guidance on the format and I can involve CELA.
As for the content itself, I don't have the list of all the 3rd part libraries our packages depend on, or what licenses they use. I will trust you have done the due diligence there :)

**Comment by nkolev92** [comment]
@karann-msft 

I guided him to use the information from https://docs.opensource.microsoft.com/using/required-notice-template.html. 

The question that needs answered is do we need to include the test libraries (that do not ship) in the 3rd party notice.

File: NOTICE.txt
Comment (nkolev92) [inline NOTICE.txt]: I don't think we need to include the bouncy castle notice as it's not shipping.

File: NOTICE.txt
Comment (nkolev92) [inline NOTICE.txt]: Same applies here.

File: NOTICE.txt
Comment (nkolev92) [inline NOTICE.txt]: ditto

File: NOTICE.txt
Comment (nkolev92) [reply NOTICE.txt]: Also here you have 

License notice**s** instead of
License notice

**Review by nkolev92 — COMMENTED** [review]
Let's run this by the https://docs.opensource.microsoft.com/ folks.

File: NOTICE.txt
Comment (dominoFire) [reply NOTICE.txt]: Typo is solved

File: NOTICE.txt
Comment (dominoFire) [reply NOTICE.txt]: Typo solved

File: NOTICE.txt
Comment (dominoFire) [reply NOTICE.txt]: Let's get some advice from the Open Source Team at Microsoft

Comment (dominoFire) [reply NOTICE.txt]: Only open source attributions shipped with NuGet needs to be reported. Thus, I'll remove the attribution for all testing assemblies.

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj]: Use $(RepositoryRootDirectory) instead of the parent directory separator.

**Review by nkolev92 — COMMENTED** [review]
Looks good, just one 1 small change needed before approval.

File: src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj]: Thanks! I'll do that

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj]: You should probably rename it to third party notices
like 
https://github.com/dotnet/cli/blob/master/THIRD-PARTY-NOTICES

I'd add the txt at the end as it's consistent with our license.txt

**Review by nkolev92 — APPROVED** [review]
LGTM. 

We discussed offline. 

ThirdPartyNotices.txt, THIRD-PARTY-NOTICES.txt are all acceptance names :)

**Pull request merged**
